### PR TITLE
Fix dataproducer caching

### PR DESCRIPTION
**Pull request by dbosen**
Make sure these boxes are checked before submitting your pull request - thank you!

- [ ] All coding styles are fulfilled. ([How to check for cs issues?](https://github.com/BurdaMagazinOrg/thunder-dev-tools/blob/master/README.md#code-style-guidelines))
- [ ] All tests are running locally. ([How to run the test?](https://github.com/thunder/thunder-distribution/blob/8.x-4.x/docs/development.md#how-to-run-the-tests))
- [ ] Necessary update hooks are provided.
- [ ] User roles have correct access for new introduced permission.
- [ ] Every thunder module has a README.md in its root. Follow [this guidelines](https://www.drupal.org/node/2181737), but we don't need every topic.
- [ ] Code is covered with well-balanced amount of inline comments.
- [ ] New features or changes are documented.

If you are really awesome, then your feature is covered by additional tests. Well done!

### CODE REVIEW

**Comment by chrfritsch** [comment]
ThunderEntitySubRequestBase is already calling `executeInRenderContext`. Maybe we can add the cacheable dependency there, instead of in every subclass.

File: modules/thunder_gqls/tests/src/Traits/ThunderGqlsTestTrait.php [hunk 1]
```diff
@@ -127,4 +138,17 @@ protected function getExpectedResponseFromFile(string $name): string {
...
+  /**
+   * Decodes a json string.
+   *
+   * @param string $json
+   *   The json string.
+   *
+   * @return mixed
+   *   The decoded json.
+   */
+  protected function jsonDecode(string $json) {
```
Comment (chrfritsch) [inline modules/thunder_gqls/tests/src/Traits/ThunderGqlsTestTrait.php:150]: Could we use Drupal\Component\Serialization\Json instead?

**Review by chrfritsch — CHANGES_REQUESTED** [review]

File: modules/thunder_gqls/tests/src/Traits/ThunderGqlsTestTrait.php (hunk 1 above)
Comment (dbosen) [reply modules/thunder_gqls/tests/src/Traits/ThunderGqlsTestTrait.php:150]: Not, if we want to keep JSON_THROW_ON_ERROR

File: modules/thunder_gqls/src/Plugin/GraphQL/DataProducer/ThunderBreadcrumb.php [hunk 2]
```diff
@@ -100,6 +100,8 @@ protected function resolve(string $path, CacheableMetadata $cacheableMetadata, F
         'title' => $text,
       ];
     }
+
+    $fieldContext->addCacheableDependency($build);
```
Comment (chrfritsch) [inline modules/thunder_gqls/src/Plugin/GraphQL/DataProducer/ThunderBreadcrumb.php:104]: Is this still necessary?

**Review by chrfritsch — APPROVED** [review]

**Pull request merged**